Fix #8841: Improve inline val error message (#8844)
lampepfl/dotty: docs/docs/reference/metaprogramming/inline.md

including _platform-specific_ extensions such as constant folding of pure
numeric computations.

An inline value must have a literal type such as `1` or `true`.
```scala
inline val four = 4
// equivalent to
inline val four: 4 = 4
```

> It is also possible to have inline vals with primitives that do not have a syntactical version such as `Short`.

Reviewer: ```suggestion
It is also possible to have inline vals of types that do not have a syntax, such as `Short(4)`.
```
